clients/src/main/java/org/apache/kafka/common/record/TimestampType.java

```diff
@@ -37,11 +37,11 @@ public byte updateAttributes(byte attributes) {
             throw new IllegalArgumentException("Cannot use NO_TIMESTAMP_TYPE in attributes");
 
         return this == CREATE_TIME ?
-            (byte) (attributes & ~Record.TIMESTAMP_TYPE_MASK) : (byte) (attributes | Record.TIMESTAMP_TYPE_MASK);
+            (byte) (attributes & ~LegacyRecord.TIMESTAMP_TYPE_MASK) : (byte) (attributes | LegacyRecord.TIMESTAMP_TYPE_MASK);
     }
 
     public static TimestampType forAttributes(byte attributes) {
-        int timestampType = (attributes & Record.TIMESTAMP_TYPE_MASK) >> Record.TIMESTAMP_TYPE_ATTRIBUTE_OFFSET;
+        int timestampType = (attributes & LegacyRecord.TIMESTAMP_TYPE_MASK) >> LegacyRecord.TIMESTAMP_TYPE_ATTRIBUTE_OFFSET;
```

Comment:
After thinking about this, it might be better to move the attribute logic into the record classes. We'll duplicate a little code, but it seems better to keep the usage of record attributes encapsulated.